### PR TITLE
HRCPP-191 Cpp client test applications should link to the static library

### PR DESCRIPTION
**Pull request by alanfx**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -196,12 +196,12 @@ set_target_properties(hotrod-static PROPERTIES COMPILE_FLAGS "-DHOTROD_DECLARE_S
 add_executable (simple test/Simple.cpp)
 set_property(TARGET simple PROPERTY INCLUDE_DIRECTORIES "${CMAKE_CURRENT_SOURCE_DIR}/include")
 set_target_properties (simple PROPERTIES COMPILE_FLAGS "${COMPILER_FLAGS} ${WARNING_FLAGS}")
-target_link_libraries (simple hotrod)
+target_link_libraries (simple hotrod-static pthread)
```
Comment (isavin) [inline CMakeLists.txt]: This breaks the build on Windows.

File: CMakeLists.txt [hunk 2]
```diff
@@ -196,12 +196,12 @@ set_target_properties(hotrod-static PROPERTIES COMPILE_FLAGS "-DHOTROD_DECLARE_S
...
 
 if (ENABLE_INTERNAL_TESTING)
     add_executable (unit_test test/Unit.cpp)
     set_target_properties (unit_test PROPERTIES COMPILE_FLAGS "${COMPILER_FLAGS} ${WARNING_FLAGS}")
-    target_link_libraries (unit_test hotrod)
+    target_link_libraries (unit_test hotrod-static pthread)
```
Comment (isavin) [inline CMakeLists.txt]: Same here.

File: CMakeLists.txt (hunk 1 above)
Comment (alanfx) [inline CMakeLists.txt]: Doh! OK, I working on this.

**Comment by alanfx** [comment]
@isavin OK, this is ready to try on all platforms! :-)

**Comment by infinispanrelease** [comment]
**FAILURE** - TeamCity HotRod C++ Client :: Pull requests monitor - WIN_S2012 [Build 203](http://ci.infinispan.org/viewLog.html?buildId=26078&buildTypeId=HotRodCpp_PullRequestsMonitorWinS2012&guest=1) for e467940ebbd541022db68011af394916bff450b2
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:01:56

Process exited with code 1

**Comment by alanfx** [comment]
@isavin I am not seeing the error from CI in my environment. I was just able to build on 32 and 64 bit Windows with these changes. Can the CI be re-run?

**Comment by isavin** [comment]
@alanfx I'm seeing the same failures as on CI when testing locally your changes.

**Comment by isavin** [comment]
@alanfx also please update the commit message to include the correct issue id https://issues.jboss.org/browse/HRCPP-191 (moved from ISPN to HRCPP).

**Comment by infinispanrelease** [comment]
**FAILURE** - TeamCity HotRod C++ Client :: Pull requests monitor - WIN_S2012 [Build 204](http://ci.infinispan.org/viewLog.html?buildId=26566&buildTypeId=HotRodCpp_PullRequestsMonitorWinS2012&guest=1) for 30cba94b5e8b0c79b901e0482e847e8085989c6c
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:01:53

Process exited with code 1

**Comment by alanfx** [comment]
@isavin ok, this works for me, but it uses the "/FORCE:MULTIPLE" linker flag on Windows.

**Comment by infinispanrelease** [comment]
**FAILURE** - TeamCity HotRod C++ Client :: Pull requests monitor - WIN_S2012 [Build 205](http://ci.infinispan.org/viewLog.html?buildId=26568&buildTypeId=HotRodCpp_PullRequestsMonitorWinS2012&guest=1) for 2690e4512559b3a96b01a0e96a4be29b712c0be5
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:01:34

Process exited with code 1

**Comment by alanfx** [comment]
CI is failing with the following error:

CUSTOMBUILD : error : M2_HOME is set to an invalid directory. [C:\BuildAgent\work\4fe618b066abd9d4\build\JniTest.vcxproj]

Not sure how to fix this. @isavin or @slaskawi?

**Comment by infinispanrelease** [comment]
**FAILURE** - TeamCity HotRod C++ Client :: Pull requests monitor - WIN_S2012 [Build 206](http://ci.infinispan.org/viewLog.html?buildId=26616&buildTypeId=HotRodCpp_PullRequestsMonitorWinS2012&guest=1) for 49f1821035f4a13ff37b0f6b435c5f981354cd0a
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:01:53

Process exited with code 1

**Comment by infinispanrelease** [comment]
**FAILURE** - TeamCity HotRod C++ Client :: Pull requests monitor - WIN_S2012 [Build 209](http://ci.infinispan.org/viewLog.html?buildId=26797&buildTypeId=HotRodCpp_PullRequestsMonitorWinS2012&guest=1) for 49f1821035f4a13ff37b0f6b435c5f981354cd0a
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:01:37

Process exited with code 1

**Comment by isavin** [comment]
@alanfx fixed the CI issues, a property was defined on the top-level project which doesn't make sense on windows.

I've opened another PR #165 with an attempt to fix the access violation in hotrod-static and I've added also the changes for HRCPP-191 in it. Please have a look to see if it's OK from your POV.

**Comment by infinispanrelease** [comment]
**FAILURE** - TeamCity HotRod C++ Client :: Pull requests monitor - WIN_S2012 [Build 214](http://ci.infinispan.org/viewLog.html?buildId=26976&buildTypeId=HotRodCpp_PullRequestsMonitorWinS2012&guest=1) for 49f1821035f4a13ff37b0f6b435c5f981354cd0a
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:06:16

Process exited with code 1

**Comment by infinispanrelease** [comment]
**SUCCESS** - TeamCity HotRod C++ Client :: Pull requests monitor - RHEL6 [Build 217](http://ci.infinispan.org/viewLog.html?buildId=26977&buildTypeId=HotRodCpp_PullRequestsMonitorRhel6&guest=1) for 49f1821035f4a13ff37b0f6b435c5f981354cd0a
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:01:11

**Comment by infinispanrelease** [comment]
**SUCCESS** - TeamCity HotRod C++ Client :: Pull requests monitor - RHEL6 [Build 218](http://ci.infinispan.org/viewLog.html?buildId=26978&buildTypeId=HotRodCpp_PullRequestsMonitorRhel6&guest=1) for 49f1821035f4a13ff37b0f6b435c5f981354cd0a
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:01:28

**Comment by alanfx** [comment]
Implemented in https://github.com/infinispan/cpp-client/pull/165

**Pull request closed without merging**